src/base/utils.js

```diff
@@ -235,10 +235,34 @@ export function removeArrayItem(arr, item) {
   }
 }
 
+// HTML5 video element recycle storage object.
+// If playback recycleVideo flag is set, it store
+// one video element for each player instance created.
+const videoStore = {}
+
+export class VideoStore {
```

Comment:
What about a Factory class which contains recycle logic, configured by core, with a `createElement(tagName)` method ?

This method always return a Zepto element with the following logic : if recycle is enabled and requested tag name is video, then check in stack if video a element is available. If one is available, remove it from stack and return it. Otherwise create a new one and return it. _(of course if requested tag name is not video, simple return new requested Zepto Element)_

Then in HTML5Video class destroy method, if recycle is enabled then put video element back to stack using a method from factory class. Also this method, could cleanup the previously added style children(s) nodes. (also cleanup attributes and classes ?)

I see no other way to perform that feature. If the logic is moved "higher" in parent playback class, the associated dom element is already created and have some events binded (_ensureElement + delegateEvents in UIObject_).

But i may be wrong and missed a proper way to perform this. I am open to any idea or hints.

PS: also if someone could give me an example case where Clappr create multiple video element (so i can test/debug it).